packages/docusaurus-plugin-ideal-image/src/plugin-ideal-image.d.ts

```diff
@@ -41,8 +41,23 @@ declare module '@docusaurus/plugin-ideal-image' {
 declare module '@theme/IdealImage' {
   import type {ComponentProps} from 'react';
 
-  export interface Props extends ComponentProps<'img'> {
-    img: any;
-  }
+  export type SrcType = {
+    width: number;
+    path?: string;
+    size?: number;
+    format?: 'webp' | 'jpeg' | 'png' | 'gif';
+  };
+
+  export type SrcImage = {
```

Comment:
Considering we don't even document those props anywhere, I'd suggest constraining the API to only what we document and not what is actually possible to pass.

Keeping a small API surface is likely to make it easier later to refactor/improve this image component

Reply:
i'm not sure what do you think we should expose in here, i actually limited it already to only what is supported by plugin

theoretically user can provide anything what is present in 
https://github.com/endiliey/react-ideal-image/blob/de4e8f0388ac3645d3f32355c79c3b6a7cc61ff3/index.d.ts#L22-L88

Reply:
hmmm, I think you are right, let's move on

the thing is, this plugin's prop is not really supposed to be crafted manually but should rather be obtained by an import/require + image loader so 🤷‍♂️  not sure how to enforce that